### ISSUE TITLE
[Win] Clicking notifications (in Action Center) does nothing

### ISSUE THREAD

**Issue by tooomm** [freaktechnik/advanced-github-notifier#118]
As in the title. Clicking a notification should open the relevant chat (and scroll to that message and highlight it).

In addition, when you miss a notification or not click it when it appears...the notifications are piled up in the Windows Notification/Action Center.
When I read the new comments via Firefox afterwards, there is still a mess of already read notifications left because they get not cleared up after reading.
The badge on the addon icon in Firefox updates correctly in that case (but it takes a while!).

Windows 10 1903
Firefox 69.0

**Comment by freaktechnik** [comment]
Yes, that's what notifications should do. However I think this is out of my control and more of an issue with the notification API Firefox provides and how that interacts with the action center.

The extension can not influence at all how the action center handles notifications, so they will just pile up there.

Yeah, the badge and panel can lag behind a bit.

**Comment by tooomm** [comment]
Hmm, there is no e.g. clear command to interact with already send notifications for exammple?

**Comment by freaktechnik** [comment]
There is a function to clear notifications, however in past tests it only worked while the actual notification was believed to be shown by Firefox (so when clicking it actually works). I think it may still make sense to start clearing notifications in case that's fixed (or works on Mac OS, for example).

**Comment by freaktechnik** [comment]
The latest version (1.6.x) will clear notifications when they are no longer active, however this will likely not fix the root issue here.

**Comment by tooomm** [comment]
>The latest version (1.6.x) will clear notifications when they are no longer active, however this will likely not fix the root issue here.

@freaktechnik Now, with 1.6.3, it looks like the notifications get cleared immediately and always right after they popped up. The extension lost functionality and is less useful now. :/

For Example:
New issues is opened.
Notification gets triggered and the extension shows a pop up.
The notification is recognized in the action center of windows while the notification stays alive (~5s?).
Native windows notification disappears.
Notification is cleared from action center immediately.

The problem here is that when you leave your pc and come back you have no clue if any notifications happened and if you missed something, because every notification stays only alive in the action center during the lifetime of the pop-up itself. Once the pop-up is gone, every hint that it ever happened is gone too.
They should only get cleared at the time you read e.g. the new issue and your extension removes the red badge from the icon in the firefox bar as well!

**Comment by freaktechnik** [comment]
Hm, I did not change anything in that regard. That must be Firefox 74 or whatever's on stable behaving differently/more consistently with what it supports.

All I added was logic to clear a notification if it was no longer returned by GitHub's notification API: https://github.com/freaktechnik/advanced-github-notifier/commit/206717c85a9b89d443283e8e60a99e0f7968e8f5

**Comment by tooomm** [comment]
Looks like we have to wait for a better integration of native notifications in windows first...

**Comment by freaktechnik** [comment]
I think this is now fixed in Firefox.
